plugin-core/src/main/java/com/salesforce/bazel/eclipse/builder/BazelErrorStreamObserver.java

```diff
@@ -0,0 +1,143 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ */
+package com.salesforce.bazel.eclipse.builder;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.LinkedList; 
+import java.util.Map;
+import java.util.Queue; 
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.IProgressMonitor;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+import com.salesforce.bazel.eclipse.abstractions.OutputStreamObserver;
+import com.salesforce.bazel.eclipse.builder.BazelMarkerSupport;
+import com.salesforce.bazel.eclipse.logging.LogHelper;
+import com.salesforce.bazel.eclipse.model.BazelBuildError;
+import com.salesforce.bazel.eclipse.model.BazelLabel;
+import com.salesforce.bazel.eclipse.model.BazelOutputParser;
+
+/**
+ * Implementation of {@link OutputStreamObserver} that observes error output and publishes errors to Problems View
+ */
+public class BazelErrorStreamObserver implements OutputStreamObserver {
+    private final IProgressMonitor monitor;
+    private final Map<BazelLabel, IProject> labelToProject;
+    private final IProject rootProject;
+    private static final BazelOutputParser outputParser = new BazelOutputParser();
+    private final Queue<String> errorLines = new LinkedList<>();
+    private static final LogHelper LOG = LogHelper.log(BazelErrorStreamObserver.class);
+    static final String UNKNOWN_PROJECT_ERROR_MSG_PREFIX = "ERROR IN UNKNOWN PROJECT: ";
+
+    public BazelErrorStreamObserver(final IProgressMonitor monitor, final Map<BazelLabel, IProject> labelToProject,
+            IProject rootProject) {
+        this.monitor = monitor;
+        this.labelToProject = labelToProject;
+        this.rootProject = rootProject;
+    }
+    
+    /**
+     * Starts the observer, clears Problems View for every project
+     */
+    @Override
+    public void startObserver() {
+        final Set<IProject> projectSet = new HashSet<>(this.labelToProject.values());
+        for (IProject project : projectSet) {
+            BazelMarkerSupport.publishToProblemsView(project, Collections.emptyList(), monitor);
+        }
+    }
+
+    /**
+     * Launch a thread to parse latest error line and publish to Problems View if applicable
+     */
+    @Override
+    public void update(String error) {
+        synchronized (this.errorLines) {
+            this.errorLines.add(error);
+        }
+        new Thread(() -> updateProblemsView()).start();
```

Comment:
There's probably an Eclipse API to run async work that we should use.

Reply:
Will look into if there is an Eclipse async API. I don't think the ```publishToProblemsView``` is async. It was causing issue for the CopyStream thread to finish. Creating a separate thread for publishing solves the issue that CopyStream thread does not finish within timeout.

Reply:
Ok, sounds good.  If no Eclipse API, you can also look at using a CompletableFuture and https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Executors.html.  Related to this, I think you don't need so synchronize as much - the errorLines can just be passed into the thread you create, I think?